soc/arm/rpi_pico/rp2/Kconfig.defconfig.series

```diff
@@ -14,4 +14,8 @@ config SOC_SERIES
 config NUM_IRQS
 	default 26
 
+config UART_RPI_PICO
+	default y
+	depends on SERIAL
```

Comment:
This is usually selected by boards `CONFIG_SERIAL=y`

Reply:
What do you mean? This only selects the rpi pico driver, if uart was enabled by `CONFIG_SERIAL=y`

Reply:
I mean that this should be removed, just add a default to the driver based on DT status (see [previous](https://github.com/zephyrproject-rtos/zephyr/pull/34835#discussion_r768726000) comment). Then boards will just add `CONFIG_SERIAL=y`.